velox/functions/prestosql/aggregates/ApproxDistinctAggregate.cpp

```diff
@@ -175,17 +174,19 @@ class ApproxDistinctAggregate : public exec::Aggregate {
             FlatVector<StringView>* result,
             vector_size_t index) {
           auto size = accumulator->serializedSize();
+          StringView serialized;
           if (StringView::isInline(size)) {
-            StringView serialized(size);
-            accumulator->serialize(indexBitLength_, serialized);
-            result->setNoCopy(index, serialized);
+            std::string buffer(size, '\0');
```

Comment:
Can we avoid heap allocation here by using `char[StringView::kInlineSize + 1]`?

Reply:
`std::string` will not heap allocate for short strings. For libstdc++ [local capacity is 15 characters](https://github.com/gcc-mirror/gcc/blob/dfa85be/libstdc%2B%2B-v3/include/bits/basic_string.h#L213)

Reply:
What's the mininum version this is supported in gcc and clang?  If it is not too far away, we still need to be careful.

Reply:
This is supported in versions 5 of both [gcc's libstdc++](https://github.com/gcc-mirror/gcc/blob/releases/gcc-5.1.0/libstdc%2B%2B-v3/include/bits/basic_string.h#L117) and [llvm's libcxx](https://github.com/llvm/llvm-project/blob/llvmorg-5.0.0/libcxx/include/string#L681). If that's not sufficient, I'm happy to use the local array instead

Reply:
There are still CentoOS 7 in the major cloud providers until next year, and the default gcc there is still 4.8.5, so we probably want to make this compatible.  This should not be a huge change anyway.